src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala

```diff
@@ -137,7 +137,53 @@ trait RawlsGroupComponent {
         
         nextLevelUp.result.flatMap { nextGroups =>
           val nextCumulativeGroups = cumulativeGroups ++ nextGroups
-          listParentGroupsRecursive(nextGroups.toSet -- nextCumulativeGroups, nextCumulativeGroups)
+          listParentGroupsRecursive(nextGroups.toSet -- cumulativeGroups, nextCumulativeGroups)
```

Comment:
yes, this was intentional. it was a bug that caused the recursion to terminate after the first step. it's out of scope for this PR but i can add a test for this function if you'd like